plugins/tactics/src/Ide/Plugin/Tactic/Types.hs

```diff
@@ -0,0 +1,149 @@
+{-# LANGUAGE DeriveFunctor                  #-}
+{-# LANGUAGE DeriveGeneric                  #-}
+{-# LANGUAGE DerivingStrategies             #-}
+{-# LANGUAGE FlexibleInstances              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving     #-}
+{-# LANGUAGE MultiParamTypeClasses          #-}
+{-# LANGUAGE TypeSynonymInstances           #-}
+
+module Ide.Plugin.Tactic.Types
+  ( module Ide.Plugin.Tactic.Types
+  , module Ide.Plugin.Tactic.Debug
+  , OccName
+  , Name
+  , Type
+  , TyVar
+  , Span
+  , Range
+  ) where
+
+import Control.Monad.Reader
+import Data.Function
+import Data.Map (Map)
+import Data.Set (Set)
+import Development.IDE.GHC.Compat
+import Development.IDE.Types.Location
+import GHC.Generics
+import Ide.Plugin.Tactic.Debug
+import OccName
+import Refinery.Tactic
+import Type
+
+
+------------------------------------------------------------------------------
+-- | A wrapper around 'Type' which supports equality and ordering.
+newtype CType = CType { unCType :: Type }
+
+instance Eq CType where
+  (==) = eqType `on` unCType
+
+instance Ord CType where
+  compare = nonDetCmpType `on` unCType
+
+
+------------------------------------------------------------------------------
+data TacticState = TacticState
+    { ts_skolems   :: [TyVar]
+    , ts_unifier   :: TCvSubst
+    , ts_used_vals :: Set OccName
+    }
```

Comment:
These should all probably be strict fields

Reply:
Done